### ISSUE TITLE
Suggestions for Colab notebook use?

### ISSUE THREAD

**Issue by metaphorz** [toshas/torch-fidelity#24]
I do not have CUDA installed since I am running an Apple imac. However, I use Google Colab for most machine learning experiments. I started a new notebook with pip install torch-fidelity, which worked fine and received:

.....
Installing collected packages: torch-fidelity
Successfully installed torch-fidelity-0.3.0

However, I don't know where the bin "fidelity" is installed in Colab space. I looked in user/local/bin and it was not there. Same issue with /usr/bin. In the notebook, "whereis fidelity" or "type -a fidelity" are not yielding anything. I also tried accessing fidelity directly from python code in the readme.md 

Any suggestions to move forward on this?

**Comment by toshas** [comment]
Colab space seems to work for me:

```
In[1]: !pip install torch-fidelity
Out[1]:
Collecting torch-fidelity
  Downloading torch_fidelity-0.3.0-py3-none-any.whl (37 kB)
Requirement already satisfied: tqdm in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (4.41.1)
Requirement already satisfied: torchvision in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (0.10.0+cu102)
Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (1.19.5)
Requirement already satisfied: scipy in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (1.4.1)
Requirement already satisfied: torch in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (1.9.0+cu102)
Requirement already satisfied: Pillow in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (7.1.2)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from torch->torch-fidelity) (3.7.4.3)
Installing collected packages: torch-fidelity
Successfully installed torch-fidelity-0.3.0

In[2]: !fidelity --input1 cifar10-train --isc --gpu 0
Out[2]:
Creating feature extractor "inception-v3-compat" with features ['logits_unbiased']
Downloading: "https://github.com/toshas/torch-fidelity/releases/download/v0.2.0/weights-inception-2015-12-05-6726825d.pth" to /root/.cache/torch/hub/checkpoints/weights-inception-2015-12-05-6726825d.pth
100% 91.2M/91.2M [00:03<00:00, 30.9MB/s]
Extracting features from input1
Downloading https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz to /root/.cache/torch/fidelity_datasets/cifar-10-python.tar.gz
170499072it [00:06, 27533449.69it/s]                   
Extracting /root/.cache/torch/fidelity_datasets/cifar-10-python.tar.gz to /root/.cache/torch/fidelity_datasets
/usr/local/lib/python3.7/dist-packages/torch/utils/data/dataloader.py:481: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  cpuset_checked))
Processing samples:   0% 0/50000 [00:00<?, ?samples/s]/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
Processing samples
Caching /root/.cache/torch/fidelity_cache/cifar10-train-inception-v3-compat-features-logits_unbiased.pt
Inception Score: 11.236778961941805 ± 0.09513778684942122
inception_score_mean: 11.23678
inception_score_std: 0.09513779

In[3]: !which fidelity
Out[3]: /usr/local/bin/fidelity

In[4]: !cat /usr/local/bin/fidelity
Out[4]: 
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import sys
from torch_fidelity.fidelity import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

**Comment by metaphorz** [comment]
You are right—I don’t know what I was trying yesterday—it may have been a mis-setting of

the runtime in Colab. Thanks.

-p

 

 

Colab space seems to work for me:
In[1]: !pip install torch-fidelity
 
Out[1]:
 
Collecting torch-fidelity
 
  Downloading torch_fidelity-0.3.0-py3-none-any.whl (37 kB)
 
Requirement already satisfied: tqdm in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (4.41.1)
 
Requirement already satisfied: torchvision in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (0.10.0+cu102)
 
Requirement already satisfied: numpy in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (1.19.5)
 
Requirement already satisfied: scipy in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (1.4.1)
 
Requirement already satisfied: torch in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (1.9.0+cu102)
 
Requirement already satisfied: Pillow in /usr/local/lib/python3.7/dist-packages (from torch-fidelity) (7.1.2)
 
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.7/dist-packages (from torch->torch-fidelity) (3.7.4.3)
 
Installing collected packages: torch-fidelity
 
Successfully installed torch-fidelity-0.3.0
 
 
 
In[2]: !fidelity --input1 cifar10-train --isc --gpu 0
 
Out[2]:
 
Creating feature extractor "inception-v3-compat" with features ['logits_unbiased']
 
Downloading: "https://github.com/toshas/torch-fidelity/releases/download/v0.2.0/weights-inception-2015-12-05-6726825d.pth" to /root/.cache/torch/hub/checkpoints/weights-inception-2015-12-05-6726825d.pth
 
100% 91.2M/91.2M [00:03<00:00, 30.9MB/s]
 
Extracting features from input1
 
Downloading https://www.cs.toronto.edu/~kriz/cifar-10-python.tar.gz to /root/.cache/torch/fidelity_datasets/cifar-10-python.tar.gz
 
170499072it [00:06, 27533449.69it/s]                   
 
Extracting /root/.cache/torch/fidelity_datasets/cifar-10-python.tar.gz to /root/.cache/torch/fidelity_datasets
 
/usr/local/lib/python3.7/dist-packages/torch/utils/data/dataloader.py:481: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
 
  cpuset_checked))
 
Processing samples:   0% 0/50000 [00:00<?, ?samples/s]/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py:718: UserWarning: Named tensors and all their associated APIs are an experimental feature and subject to change. Please do not use them for anything important until they are released as stable. (Triggered internally at  /pytorch/c10/core/TensorImpl.h:1156.)
 
  return torch.max_pool2d(input, kernel_size, stride, padding, dilation, ceil_mode)
 
Processing samples
 
Caching /root/.cache/torch/fidelity_cache/cifar10-train-inception-v3-compat-features-logits_unbiased.pt
 
Inception Score: 11.236778961941805 ± 0.09513778684942122
 
inception_score_mean: 11.23678
 
inception_score_std: 0.09513779
 
 
 
In[3]: !which fidelity
 
Out[3]: /usr/local/bin/fidelity
 
 
 
In[4]: !cat /usr/local/bin/fidelity
 
Out[4]: 
 
#!/usr/bin/python3
 
# -*- coding: utf-8 -*-
 
import re
 
import sys
 
from torch_fidelity.fidelity import main
 
if __name__ == '__main__':
 
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
 
    sys.exit(main())
